### PR TITLE
PIM-3902: Add Remove action and CategoryFieldRemover

### PR DESCRIPTION
**Pull request by willy-ahva**
| Q                    | A
| -------------------- | ---
| Bug fix?             | n
| New feature?         | ?
| BC breaks?           | n
| CI currently passes? | 
| Tests pass?          | y
| Scenarios pass?      | y
| Checkstyle issues?*  | n
| PMD issues?**        | n
| Changelog updated?   | n
| Fixed tickets        | PIM-3902
| DB schema updated?   | n
| Migration script?    | n
| Doc PR               |

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Updater/Remover/RemoverRegistry.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    public function getAttributeRemover(AttributeInterface $attribute)
+    {
+        foreach ($this->attributeRemovers as $remover) {
+            if ($remover->supportsAttribute($attribute)) {
+                return $remover;
+            }
+        }
+
+        return null;
```
Comment (solivier) [inline src/Pim/Bundle/CatalogBundle/Updater/Remover/RemoverRegistry.php]: useless

File: src/Pim/Bundle/CatalogBundle/Updater/Remover/RemoverRegistry.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    public function getFieldRemover($field)
+    {
+        foreach ($this->fieldRemovers as $remover) {
+            if ($remover->supportsField($field)) {
+                return $remover;
+            }
+        }
+
+        return null;
```
Comment (solivier) [inline src/Pim/Bundle/CatalogBundle/Updater/Remover/RemoverRegistry.php]: useless

Comment (willy-ahva) [inline src/Pim/Bundle/CatalogBundle/Updater/Remover/RemoverRegistry.php]: It is everywhere in all others Registry so I put it.

File: src/Pim/Bundle/CatalogBundle/Updater/Remover/RemoverRegistry.php (hunk 1 above)
Comment (willy-ahva) [inline src/Pim/Bundle/CatalogBundle/Updater/Remover/RemoverRegistry.php]: Same : It is everywhere in all others Registry so I put it.

File: src/Pim/Bundle/CatalogBundle/Updater/Remover/RemoverRegistry.php (hunk 2 above)
Comment (solivier) [inline src/Pim/Bundle/CatalogBundle/Updater/Remover/RemoverRegistry.php]: not a reason ;)

File: src/Pim/Bundle/CatalogBundle/Updater/ProductUpdater.php [hunk 3]
```diff
@@ -132,6 +138,31 @@ public function copyData(
...
+    {
+        $attribute = $this->getAttribute($field);
+        if (null !== $attribute) {
+            $remover = $this->removerRegistry->getAttributeRemover($attribute);
+        } else {
+            $remover = $this->removerRegistry->getFieldRemover($field);
+        }
+
+        if (null === $remover) {
+            throw new \LogicException(sprintf('No remover found for field "%s"', $field));
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Updater/ProductUpdater.php]: for field or attribute "%s"

File: src/Pim/Bundle/CatalogBundle/Updater/Remover/CategoryFieldRemover.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+            if (null === $category) {
+                throw InvalidArgumentException::expected(
+                    $field,
+                    'existing category code',
+                    'remover',
+                    'category',
+                    $categoryCode
+                );
+            } else {
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Updater/Remover/CategoryFieldRemover.php]: useless else

File: src/Pim/Bundle/CatalogBundle/Updater/Remover/RemoverRegistry.php (hunk 2 above)
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Updater/Remover/RemoverRegistry.php]: not useless

Comment (solivier) [inline src/Pim/Bundle/CatalogBundle/Updater/Remover/RemoverRegistry.php]: so we need to add it everywhere

File: src/Pim/Bundle/CatalogBundle/Updater/Remover/RemoverRegistry.php (hunk 1 above)
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Updater/Remover/RemoverRegistry.php]: same that my other comment
@willy-ahva your answer is not valid, you should ask yourself or the others why it's here

File: src/Pim/Bundle/CatalogBundle/Updater/Setter/SetterRegistryInterface.php [hunk 5]
```diff
@@ -27,7 +27,7 @@ public function register(SetterInterface $setter);
      *
      * @param string $field the field
      *
-     * @return SetterInterface|null
+     * @return FieldSetterInterface|null
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Updater/Setter/SetterRegistryInterface.php]: :+1:

**Pull request merged**